src/components/variable-picker.ts

```diff
@@ -83,9 +88,33 @@ export class SchedulerVariablePicker extends LitElement {
     `;
   }
 
+  renderTimeVariable() {
+    if (!this.hass || !this.variable) {
+      console.warn(`${this.renderTimeVariable.name}() not rendering: undefined references`);
+      return html``;
+    }
+    const variable = this.variable as TimeVariable;
+    const value = this.value;
+
+    return html`
+      <ha-time-input
+        .enableSecond=${variable.enable_seconds}
+        .value=${value}
+        .locale=${this.hass.locale}
+        @value-changed=${this.listVariableUpdated}
+      ></ha-time-input>
+      <div class="key">${variable.enable_seconds ? 'Hours:Minutes:Seconds' : 'Hours:Minutes'}</div>
+    `;

```

Comment:
The choice of the `<ha-time-input>` component can be argued. My reasoning for not using the `<time-picker>` component (which is used to set / edit the scheduler time) was simply to visually differentiate between the two time sections (see other comments). Users would be used to the `<time-picker>` component for setting the scheduler time and they would also be used to the `<ha-time-input>` component for editing HA time entities.